### PR TITLE
[POC] Multiple document managers.

### PR DESCRIPTION
**Pull request by dantleech**
This PR introduces the possiblity to have multiple document managers and access them via. a `DocumentManagerRegistry`.

This allows us to access more than one PHPCR workspace or backend in the same request.

WARNING: This is using a POC branch of the document manager, and as such most of the code in this PR is liable to change if we implement a different strategy in the document manager.

See DocumentManager PR for more details: https://github.com/sulu-io/sulu-document-manager/pull/66

Notes

- We move the PHPCR configuration out of the `SuluCoreBundle` to the `DocumentManagerBundle`.

### CODE REVIEW

File: src/Sulu/Bundle/CoreBundle/DependencyInjection/SuluCoreExtension.php [hunk 1]
```diff
@@ -31,39 +31,6 @@ class SuluCoreExtension extends Extension implements PrependExtensionInterface
...
-        $parameterBag = $container->getParameterBag();
-        $configs = $parameterBag->resolveValue($configs);
-        $config = $this->processConfiguration(new Configuration(), $configs);
-
-        if (isset($config['phpcr'])) {
-            $phpcrConfig = $config['phpcr'];
-
-            // TODO: Workaround for issue: https://github.com/doctrine/DoctrinePHPCRBundle/issues/178
-            if (!isset($phpcrConfig['backend']['check_login_on_server'])) {
-                $phpcrConfig['backend']['check_login_on_server'] = false;
```
Comment (dantleech) [inline src/Sulu/Bundle/CoreBundle/DependencyInjection/SuluCoreExtension.php]: This stuff has no right to be in the CoreBundle .. removed *for now* and using the `doctrine_phpcr` config directly.

Comment (danrot) [inline src/Sulu/Bundle/CoreBundle/DependencyInjection/SuluCoreExtension.php]: I would put that into the `prepend` method of the `SuluContentExtension`.

Comment (dantleech) [inline src/Sulu/Bundle/CoreBundle/DependencyInjection/SuluCoreExtension.php]: Nope, its for the document manager bundle ..

**Comment by dantleech** [comment]
Includes #2056 will rebase this out after that PR is merged.

File: src/Sulu/Bundle/DocumentManagerBundle/Bridge/DocumentInspectorFactory.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+use Sulu\Component\DocumentManager\DocumentManagerContext;
+use Sulu\Component\DocumentManager\DocumentInspectorFactoryInterface;
+use Sulu\Component\DocumentManager\PathSegmentRegistry;
+use Sulu\Component\DocumentManager\NamespaceRegistry;
+use Sulu\Component\DocumentManager\MetadataFactoryInterface;
+use Sulu\Bundle\DocumentManagerBundle\Bridge\PropertyEncoder;
+use Sulu\Component\Webspace\Manager\WebspaceManagerInterface;
+use Sulu\Component\Content\Metadata\Factory\StructureMetadataFactoryInterface;
+
+class DocumentInspectorFactory implements DocumentInspectorFactoryInterface
```
Comment (dantleech) [inline src/Sulu/Bundle/DocumentManagerBundle/Bridge/DocumentInspectorFactory.php]: doc

File: src/Sulu/Bundle/DocumentManagerBundle/Bridge/DocumentManagerRegistry.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Sulu\Bundle\DocumentManagerBundle\Bridge;
+
+use Sulu\Component\DocumentManager\DocumentManagerRegistryInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+class DocumentManagerRegistry implements DocumentManagerRegistryInterface
```
Comment (dantleech) [inline src/Sulu/Bundle/DocumentManagerBundle/Bridge/DocumentManagerRegistry.php]: doc

File: src/Sulu/Component/Rest/Tests/Functional/ListBuilder/Metadata/cache/stdClass.php [hunk 4]
```diff
@@ -0,0 +1 @@
+<?php return unserialize('a:0:{}');
```
Comment (dantleech) [inline src/Sulu/Component/Rest/Tests/Functional/ListBuilder/Metadata/cache/stdClass.php]: rm

File: src/Sulu/Bundle/ContentBundle/Tests/Functional/Command/CleanupHistoryCommandTest.php [hunk 5]
```diff
@@ -305,6 +305,9 @@ private function runCommandTest($webspaceKey, $locale, $dryRun, $basePath, $urls
                 '--base-path' => $basePath,
             ]
         );
+        foreach ($result as $res) {
+            var_dump($res->getId());
+        }
```
Comment (dantleech) [inline src/Sulu/Bundle/ContentBundle/Tests/Functional/Command/CleanupHistoryCommandTest.php]: rm

File: src/Sulu/Bundle/DocumentManagerBundle/Tests/Unit/Bridge/DocumentManagerRegistryTest.php [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+use Sulu\Bundle\DocumentManagerBundle\Bridge\DocumentManagerRegistry;
+use Sulu\Component\DocumentManager\DocumentManagerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+class DocumentManagerRegistryTest extends \PHPUnit_Framework_TestCase
+{
+    private $registry;
+    private $container;
+    private $manager;
```
Comment (dantleech) [inline src/Sulu/Bundle/DocumentManagerBundle/Tests/Unit/Bridge/DocumentManagerRegistryTest.php]: docs

File: src/Sulu/Bundle/DocumentManagerBundle/Tests/Unit/Command/InitializeCommandTest.php [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+class InitializeCommandTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var mixed
+     */
+    private $questionHelper;
+
+    /**
+     * @var mixed
```
Comment (dantleech) [inline src/Sulu/Bundle/DocumentManagerBundle/Tests/Unit/Command/InitializeCommandTest.php]: types

File: src/Sulu/Bundle/ContentBundle/DocumentManager/ContentInitializer.php [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+use Sulu\Component\PHPCR\NodeTypes\Path\PathNodeType;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Initialize the namespaces and content types for Sulu Content Documents.
+ */
+class ContentInitializer implements InitializerInterface
+{
+    /**
+     * @var DocumentRegistry
```
Comment (dantleech) [inline src/Sulu/Bundle/ContentBundle/DocumentManager/ContentInitializer.php]: ConectionRegi

File: src/Sulu/Bundle/CoreBundle/DependencyInjection/SuluCoreExtension.php [hunk 9]
```diff
@@ -37,25 +37,21 @@ public function prepend(ContainerBuilder $container)
...
         if (isset($config['phpcr'])) {
-            $phpcrConfig = $config['phpcr'];
-
-            // TODO: Workaround for issue: https://github.com/doctrine/DoctrinePHPCRBundle/issues/178
-            if (!isset($phpcrConfig['backend']['check_login_on_server'])) {
-                $phpcrConfig['backend']['check_login_on_server'] = false;
-            }
-
-            foreach ($container->getExtensions() as $name => $extension) {
+            foreach (array_keys($container->getExtensions()) as $name) {
```
Comment (dantleech) [inline src/Sulu/Bundle/CoreBundle/DependencyInjection/SuluCoreExtension.php]: Is there a `getExtensionNames` method?

File: src/Sulu/Bundle/DocumentManagerBundle/Bridge/DocumentInspectorFactory.php [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+     * @var DocumentManagerContext
+     */
+    private $context;
+
+    private $namespaceRegistry;
+    private $metadataFactory;
+    private $structureFactory;
+    private $encoder;
+    private $webspaceManager;
+    private $pathSegmentRegistry;
```
Comment (dantleech) [inline src/Sulu/Bundle/DocumentManagerBundle/Bridge/DocumentInspectorFactory.php]: `@var`

File: src/Sulu/Bundle/DocumentManagerBundle/Tests/app/config/sulu.yml [hunk 11]
```diff
@@ -0,0 +1,2 @@
+sulu_core:
+    webspace:
```
Comment (dantleech) [inline src/Sulu/Bundle/DocumentManagerBundle/Tests/app/config/sulu.yml]: rm

File: src/Sulu/Bundle/DocumentManagerBundle/Tests/app/Resources/pages/default.xml [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+            <meta>
+                <title lang="de">Tier</title>
+                <title lang="en">Animals</title>
+            </meta>
+
+            <tag name="sulu.rlp.part"/>
+        </property>
+
+        <property name="url" type="resource_locator" mandatory="true">
+            <meta>
```
Comment (dantleech) [inline src/Sulu/Bundle/DocumentManagerBundle/Tests/app/Resources/pages/default.xml]: Necessary to add these bootstrap files currently unless we refactor even more stuff. But perhaps we could remove th efunctional test from this bundle.

File: src/Sulu/Bundle/CoreBundle/DependencyInjection/SuluCoreExtension.php (hunk 9 above)
Comment (dantleech) [inline src/Sulu/Bundle/CoreBundle/DependencyInjection/SuluCoreExtension.php]: No

File: src/Sulu/Bundle/DocumentManagerBundle/Bridge/DocumentInspectorFactory.php [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+     */
+    public function attachContext(DocumentManagerContext $context)
+    {
+        $this->context = $context;
+    }
+
+    /**
+     * Create a new instance of the inspector.
+     *
+     * TODO: Cache me.
```
Comment (dantleech) [inline src/Sulu/Bundle/DocumentManagerBundle/Bridge/DocumentInspectorFactory.php]: Cache me

**Comment by dantleech** [comment]
Closing this, will open a new PR from my fork.

**Pull request closed without merging**